opentype-shaping-arabic-general.md

```diff
@@ -252,51 +252,23 @@ JOIN_CAUSING:
     preceding character:
 	  - `isol` becomes `init`
 	  - `fina` becomes `medi`
-	  - `init` remains `init`
-	  - `medi` remains `medi`
   - If the current character's JOINING_TYPE is RIGHT, DUAL, or
     JOIN_CAUSING, tag the current character for `fina`, then update
     the tag for the preceding character:
 	  - `isol` becomes `init`
 	  - `fina` becomes `medi`
-	  - `init` remains `init`
-	  - `medi` remains `medi`
-  - If the current character's JOINING_TYPE is LEFT or NON_JOINING,
-    tag the current character for `isol`, then update
-    the tag for the preceding character:
-	  - `medi` becomes `fina`
-	  - `init` becomes `isol`
-	  - `fina` remains `fina`
-	  - `isol` remains `isol`
 
-If the preceding character's JOINING_TYPE is RIGHT or NON_JOINING:
-  - Tag the current character for `isol`, then update the tag for the
-    preceding character:
-	  - `medi` becomes `fina`
-	  - `init` becomes `isol`
-	  - `fina` remains `fina`
-	  - `isol` remains `isol`
-	  
+Otherwise, tag the current character for `isol`.
+
 After testing the final character of the word, if the text is in
-`<syrc>` and current (final) character is "Alaph", perform an additional test:
-  - If the preceding character's JOINING_GROUP is DALATH_RISH,
-    tag the current character for `fin3`, then update the tag for the
-    preceding character:
-	  - `medi` becomes `fina`
-	  - `init` becomes `isol`
-	  - `fina` remains `fina`
-	  - `isol` remains `isol`
-  - If the preceding character's JOINING_GROUP is not DALATH_RISH, tag
-    the current character for `fin2`, then update the tag for the
-    preceding character:
-	  - `medi` becomes `fina`
-	  - `fin2` becomes `fina`
-	  - `fin3` becomes `fina`
-	  - `init` becomes `isol`
-	  - `fina` remains `fina`
-	  - `isol` remains `isol`
-	  - `med2` remains `med2`
+`<syrc>` and if the last character that is not JOINING_TYPE_TRANSPARENT or
+JOINING_TYPE_NON_JOINING is "Alaph", perform an additional test:
```

Comment:
This is a good catch, in that the existing text is not explicit enough about how to treat `TRANSPARENT` and `NON_JOINING`.